internal/cluster/cluster.go

```diff
@@ -777,10 +778,30 @@ func (m *Manager) DeleteClusterFiles(ctx context.Context, c *common.Cluster, obj
 	return nil
 }
 
-func (m Manager) PermanentClustersDeletion(ctx context.Context, olderThen strfmt.DateTime, objectHandler s3wrapper.API) error {
+func (m Manager) InactiveClusterDeregister(ctx context.Context, inactiveSince strfmt.DateTime) error {
```

Comment:
I think this functionality should be at the inventory API level, if you clean some clusters you need to use the full flow.

By using only the cluster API for deregister (instead of inventory API) you are missing:
1. cleaning the cluster subscription in AMS (OCM integration)
2. deleting the DNS record

Reply:
That's a good point - the question is if we should use the inventory API level or should we move the cleaning the cluster subscription in AMS (OCM integration) & deleting the DNS record into cluster API. @filanov @avishayt WDYT?